File: 1-Iniciante/ex1176.c
#include <stdio.h>

int fibonacci(int x);

int main(){
    /*variaveis*/
    int t, n, i, fib;

    /*entrada & processamento*/
    scanf("%d", &t);

    for(i = 0; i < t; i++){
        scanf("%d", &n);
        fib = fibonacci(n);
        printf("Fib(%d) = %d\n", n, fib); //saida
    }
    return 0;
}   

int fibonacci(int x){
    /*variaveis*/
    int i, fib[60];

    /*processamento*/
    fib[0] = 0;
    fib[1] = 1;

    if(x > 1){
        for(i = 2; i < 60; i++){
            fib[i] = fib[i - 1] + fib[i - 2];
        }
    }

    /*retorno*/
    return fib[x];
}

File: 1-Iniciante/ex1176.js
// entrada
let input = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n').map(v => parseInt(v)); // adicione uma barra "/" antes do "dev" para ser aceito no beecrowd
let n = input.shift();

// processamento & saida
for(let i = 0; i < n; i++){
    let valor = input.shift();
    let enesimo_fib = fibonacci(valor);
    console.log(`Fib(${valor}) = ${enesimo_fib}`); 
}

function fibonacci(n){
    if(n === 0) return 0;
    else if(n === 1) return 1;
    
    let n_menos_2 = 0;
    let n_menos_1 = 1;
    let atual = n_menos_2 + n_menos_1;
    for(let i = 3; i <= n; i++){
        n_menos_2 = n_menos_1;
        n_menos_1 = atual;
        atual = n_menos_2 + n_menos_1;
    }
    return atual;
}